cache/filecache/filecache.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package filecache provides an implementation of prototransform.Cache
+// that is based on the file system. Cached data is stored in and loaded
+// from files, with cache keys being used to form the file names.
+//
+// This is the simplest form of caching when sharing cache results is
+// not needed and the workload has a persistent volume (e.g. if the
+// application restarts, it will still have access to the same cache
+// files written before it restarted).
+package filecache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/bufbuild/prototransform"
+)
+
+type Config struct {
+	// Required: the folder in which cached files live.
+	Path string
```

Comment:
That seems a little odd -- it would only make sense in local dev situations. In production, this thing is likely running in a container as root (and putting the cache in root's home dir doesn't seem right) or as an unprivileged user that doesn't even have a home dir.